### PR TITLE
#1236: SPUDownloaderSession main thread

### LINKED ISSUE #1236: SPUDownloaderSession is invoking callbacks on a background thread and Sparkle is not designed for this
In 1.19. SPUDownloaderSession allocs a new background operation queue for delegate callbacks, these callbacks then message sparkles own callbacks on the background queue and it ends up causing GUI code and worse host app code to run on a background thread. Sparkle itself seems to be built on the old assumption that it would receive messages on the main thread and host apps are also going to assume they will receive delegate messages on the main thread.

Instead of trying to play whack-a-mole and fix each assumption in Sparkle (including all host app delegate messages) SPUDownloaderSession should just use the main thread operation queue.

Here's an example. This is actually within Sparkle itself, but due to the window server coalescing GUI updates host app GUI updates end up on the background thread too. We have asserts in a our code that assumes the main thread.

```
Process 38484 stopped
* thread #3, queue = 'NSOperationQueue 0x11bb182c0 (QOS: UNSPECIFIED)', stop reason = EXC_BREAKPOINT (code=EXC_I386_BPT, subcode=0x0)
    frame #0: 0x00000001000731bf HOST`+[PSUITheme currentTheme](self=PSUITheme, _cmd="currentTheme") at PSUITheme.m:112
   109 	
   110 	+ (PSUITheme*)currentTheme
   111 	{
-> 112 	    PSASSERT_MAIN();
   113 	    return (_themeStack.lastObject);
   114 	}
   115 	
Target 0: HOST  stopped.
(lldb) bt
* thread #3, queue = 'NSOperationQueue 0x11bb182c0 (QOS: UNSPECIFIED)', stop reason = EXC_BREAKPOINT (code=EXC_I386_BPT, subcode=0x0)
  * frame #0: 0x00000001000731bf HOST `+[PSUITheme currentTheme](self=PSUITheme, _cmd="currentTheme") at PSUITheme.m:112
    frame #1: 0x0000000100270ec4 HOST `-[DGStatusItemView(self=0x000000010f15ba80, _cmd="drawRect:", dirtyRect=(origin = (x = 0, y = 0), size = (width = 192, height = 146))) drawRect:] at DGStatusItemView.m:844
    frame #2: 0x00007fff4e660c21 AppKit`_NSViewDrawRect + 83
    frame #3: 0x00007fff4e66deb9 AppKit`-[NSView(NSInternal) _recursive:displayRectIgnoringOpacity:inGraphicsContext:shouldChangeFontReferenceColor:] + 1578
    frame #4: 0x00007fff4e66d889 AppKit`-[NSView(NSInternal) _recursive:displayRectIgnoringOpacity:inContext:shouldChangeFontReferenceColor:] + 56
    frame #5: 0x00007fff4deff46c AppKit`__46-[NSView(NSLayerKitGlue) drawLayer:inContext:]_block_invoke + 289
    frame #6: 0x00007fff4deff000 AppKit`-[NSView(NSLayerKitGlue) _drawViewBackingLayer:inContext:drawingHandler:] + 1577
    frame #7: 0x00007fff4defe9d1 AppKit`-[NSView(NSLayerKitGlue) drawLayer:inContext:] + 77
    frame #8: 0x00007fff5b90dad5 QuartzCore`CABackingStoreUpdate_ + 279
    frame #9: 0x00007fff5b90d9b4 QuartzCore`___ZN2CA5Layer8display_Ev_block_invoke + 54
    frame #10: 0x00007fff5b90d1b7 QuartzCore`-[CALayer _display] + 1830
    frame #11: 0x00007fff4defd996 AppKit`_NSBackingLayerDisplay + 556
    frame #12: 0x00007fff4def280d AppKit`-[_NSViewBackingLayer display] + 242
    frame #13: 0x00007fff5b8fed3b QuartzCore`CA::Layer::display_if_needed(CA::Transaction*) + 633
    frame #14: 0x00007fff5b8fe7f9 QuartzCore`CA::Layer::layout_and_display_if_needed(CA::Transaction*) + 35
    frame #15: 0x00007fff5b8fd894 QuartzCore`CA::Context::commit_transaction(CA::Transaction*) + 326
    frame #16: 0x00007fff5b8fd43d QuartzCore`CA::Transaction::commit() + 487
    frame #17: 0x00007fff4ded802c AppKit`NSPerformVisuallyAtomicChange + 166
    frame #18: 0x00007fff4e6854ff AppKit`-[NSWindow _doWindowOrderOutWithWithKeyCalc:forCounter:orderingDone:docWindow:] + 85
    frame #19: 0x00007fff4e6869f2 AppKit`-[NSWindow _reallyDoOrderWindowOutRelativeTo:findKey:forCounter:force:isModal:] + 446
    frame #20: 0x00007fff4dedfcf0 AppKit`-[NSWindow _reallyDoOrderWindow:relativeTo:findKey:forCounter:force:isModal:] + 172
    frame #21: 0x00007fff4dedeb69 AppKit`-[NSWindow _doOrderWindow:relativeTo:findKey:forCounter:force:isModal:] + 287
    frame #22: 0x00007fff4dede9cf AppKit`-[NSWindow orderWindow:relativeTo:] + 169
    frame #23: 0x00007fff4e6840da AppKit`-[NSWindow _finishClosingWindow] + 479
    frame #24: 0x00007fff4e024264 AppKit`-[NSWindow _close] + 378
    frame #25: 0x0000000108c2f260 Sparkle`-[SUUserInitiatedUpdateDriver closeCheckingWindow](self=0x0000000100e039b0, _cmd=<unavailable>) at SUUserInitiatedUpdateDriver.m:32 [opt]
    frame #26: 0x0000000108c2f707 Sparkle`-[SUUserInitiatedUpdateDriver appcastDidFinishLoading:](self=0x0000000100e039b0, _cmd=<unavailable>, ac=<unavailable>) at SUUserInitiatedUpdateDriver.m:67 [opt]
    frame #27: 0x0000000108c0270f Sparkle`-[SUAppcast downloaderDidFinishWithTemporaryDownloadData:](self=0x000000011bb83e40, _cmd=<unavailable>, downloadData=<unavailable>) at SUAppcast.m:120 [opt]
    frame #28: 0x0000000108c1be63 Sparkle`-[SPUDownloader downloadDidFinishWithData:](self=0x0000000100c3d890, _cmd=<unavailable>, data=<unavailable>) at SPUDownloader.m:89 [opt]
    frame #29: 0x0000000108c1013b Sparkle`-[SPUDownloaderSession downloadDidFinish](self=0x0000000100c3d890, _cmd=<unavailable>) at SPUDownloaderSession.m:148 [opt]
    frame #30: 0x0000000108c0f9a6 Sparkle`-[SPUDownloaderSession URLSession:downloadTask:didFinishDownloadingToURL:](self=0x0000000100c3d890, _cmd=<unavailable>, session=<unavailable>, downloadTask=<unavailable>, location=<unavailable>) at SPUDownloaderSession.m:77 [opt]
    frame #31: 0x00007fff4f8eb2d4 CFNetwork`__82-[NSURLSession delegate_downloadTask:didFinishDownloadingToURL:completionHandler:]_block_invoke + 38
```

### PR DESCRIPTION
**Pull request by bdb**
Fix #1236

### CODE REVIEW

File: Sparkle/SPUDownloaderSession.m [hunk 1]
```diff
@@ -29,7 +29,7 @@ @implementation SPUDownloaderSession
 
 - (void)startDownloadWithRequest:(SPUURLRequest *)request
 {
-    NSOperationQueue *queue = [[NSOperationQueue alloc] init];
+    NSOperationQueue *queue = [NSOperationQueue mainQueue];
     queue.maxConcurrentOperationCount = 1;
```
Comment (kornelski) [inline Sparkle/SPUDownloaderSession.m]: `maxConcurrentOperationCount` probably shouldn't be changed on the main thread

**Comment by kornelski** [comment]
Thanks for the PR. We have indeed made the process more multithreaded, and some callbacks may still need fixing

**Comment by bdb** [comment]
Yep. I removed it. (Rebased history.)

**Comment by kornelski** [comment]
Thanks

**Pull request merged**